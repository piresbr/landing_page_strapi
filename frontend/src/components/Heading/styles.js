import styled, { css } from 'styled-components';

//defino as cores do componente aqui, puxadas direto do theme.js
//themeDark definido dentro do componente, a troca de cores ocorre dentro do stories.jsx
export const Title = styled.h1`
  ${({ theme, themeDark }) => css`
    background-color: ${themeDark ? theme.colors.dark : theme.colors.light};
    color: ${themeDark ? theme.colors.light : theme.colors.dark};
    padding: ${theme.spacings.md};
  `}
`;
