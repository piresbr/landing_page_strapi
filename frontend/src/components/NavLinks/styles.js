import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.mediaQuery.md} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
