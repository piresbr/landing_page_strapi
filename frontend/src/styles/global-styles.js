//Ã© como se fosse um css generico, serve pra tudo
import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  body {
    /* background-color: red; */
  }
`;
