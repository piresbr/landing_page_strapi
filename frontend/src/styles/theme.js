//todas as propriedades do tema
export const theme = {
  colors: {
    base: '#000',
    white: '#ffffff',
    dark: '#0a1128',
    light: '#fff',
    primary: '#dedede',
    secondary: '#dc143c',
    mediumGray: '#DDDDDD',
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      sm: '1rem',
      md: '1.5rem',
      lg: '2rem',
      xl: '3rem',
      xxl: '4rem',
    }
  },
  spacings: {
    sm: '1rem',
    md: '1.5rem',
    lg: '2rem',
    xl: '3rem',
    xxl: '4rem',
  },
  mediaQuery: {
    sm: '(max-width: 320px)',
    md: '(min-width: 321px) and (max-width: 767px)',
    lg: '(min-width: 768px) and (max-width: 1279px)',
    xl: '(min-width: 1280px) and (max-width: 1400px)',
    xl: '(min-width: 1401px)',
  }
};
