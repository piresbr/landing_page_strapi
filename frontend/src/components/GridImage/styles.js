import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    min-height: 310px;
    object-fit: cover;
    border-radius: 5px;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
