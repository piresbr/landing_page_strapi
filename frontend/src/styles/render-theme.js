import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

//renderizando todas as configuraÃ§Ãµes do theme para todos os componentes
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
