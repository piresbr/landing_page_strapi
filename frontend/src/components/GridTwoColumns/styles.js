import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.mediaQuery.md} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xxl};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.mediaQuery.md} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
