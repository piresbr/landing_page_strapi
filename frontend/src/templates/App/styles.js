import styled from 'styled-components';
import { theme } from '../../styles/theme'

export const Wrapper = styled.div`background-color: ${theme.colors.mainBg}`;
