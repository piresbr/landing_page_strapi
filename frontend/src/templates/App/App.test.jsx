import React from 'react';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import Home from '.';
import { theme } from '../../styles/theme';

describe('<Home />', () => {
  test('renders learn react link', () => {
    renderTheme(<Home />);
  });
});
