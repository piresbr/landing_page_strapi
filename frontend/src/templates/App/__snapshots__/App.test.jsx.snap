// Vitest Snapshot v1

exports[`renders learn react link 1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    Hello
  </h1>
</div>
`;
